doc/guides/device_mgmt/smp_groups/smp_group_8.rst

```diff
@@ -0,0 +1,213 @@
+.. _mcumgr_smp_group_8:
+
+File management
+###############
+
+The file management group provides commands that allow to upload and download files
+to/from a device.
+
+File download
+*************
+
+Command allows to download contents of an existing file from specified path
+at a target device.  The command is stateless and mcumgr does not hold file
+in open state after response to the command is issued, instead a client
+application is supposed to keep track of data it has already downloaded,
+and issue subsequent requests, with modified offset, to gather entire file.
+Request does not carry size of requested chunk, the size is specified
+by application itself.
+Mcumgr server side re-opens a file for each subsequent request, and current
+specification does not provide means to identify subsequent requests as
+belonging to specified download session.  This means that the  file is not
+locked in any way or exclusively owned by mcumgr, for the time of download
+session, and may change between requests or even be removed.
+
+File download request
+=====================
+
+File download request header
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``8``        |  ``0``         |
+    +--------+--------------+----------------+
+
+CBOR Payload of request command:
+
+.. code-block:: none
+
+    {
+        (str)"off" :  (uint)
+        (str)"name" : (str)
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset to start download at                       |
+    +-----------------------+---------------------------------------------------+
+    | "name"                | absolute path to a file                           |
+    +-----------------------+---------------------------------------------------+
+
+File download response
+======================
+
+File download response header
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``1``  | ``8``        |  ``0``         |
+    +--------+--------------+----------------+
+
+CBOR payload of successful response command:
+
+.. code-block:: none
+
+    {
+        (str)"off"      : (uint)
+        (str)"data"     : (str)
+        (str)"rc"       : (int)0
```

Comment:
missing len field (present when off = 0)

Reply:
Unresolved

Reply:
> (str)"rc"       : (int)0

0 at the end?